iroha_p2p/src/network.rs

```diff
@@ -1,34 +1,44 @@
-use std::{collections::HashMap, io, net::SocketAddr};
+use std::{
+    collections::HashMap,
+    fmt::{Debug, Formatter},
+    io,
+    net::SocketAddr,
+};
 
 use async_stream::stream;
 use futures::Stream;
-use iroha_actor::{broker::Broker, Actor, Addr, Context, ContextHandler, Handler};
+use iroha_actor::{
+    broker::{Broker, BrokerMessage},
+    Actor, Addr, Context, ContextHandler, Handler,
+};
 use iroha_logger::{info, warn};
 use parity_scale_codec::{Decode, Encode};
 use tokio::net::{TcpListener, TcpStream};
 use ursa::{encryption::symm::Encryptor, kex::KeyExchangeScheme};
 
 use crate::{
-    peer::{Peer, PeerId, State},
+    peer::{Peer, PeerId},
     Error,
 };
 
 /// Main network layer structure, that is holding connections, called [`Peer`]s.
-#[derive(Debug)]
 pub struct Network<T, K, E>
 where
-    T: Encode + Decode + Send + Clone + 'static,
+    T: Encode + Decode + BrokerMessage + Send + Clone + 'static,
     K: KeyExchangeScheme + Send + 'static,
     E: Encryptor + Send + 'static,
 {
```

Comment:
Sorry for not mentioning it earlier but do we really need Network to be generic over KeyExchangeScheme and Encryptor? I think not, as we are not planning to have several different versions of Network with different schemes and encryptors in our codebase.

Better to remove generics here to my mind.

Reply:
If you think it should be done in a separate PR, you can create an issue for it.

Reply:
I think the same, but @i1i1 insisted on that :(

Reply:
I think it is better to be generic as there are many use cases as public net and cbdc.

Reply:
Let's discuss it on status tomorrow I guess

Reply:
It was decided to add type aliases.